### ISSUE TITLE
For nested tests, the report uses an incomplete test name

### ISSUE THREAD

**Issue by BorzdeG** [kotest/kotest#2730]
kotest version: 5.0.1

```
internal class MyTests: DescribeSpec({
  describe("level 1.1"){
    it("level 2 - case 1"){}
    it("level 2 - case 2"){}
  }
  describe("level 1.2"){
    it("level 2 - case 1"){}
    it("level 2 - case 2"){}
  }
})
```

the report gives the following list of tests:
```
MyTests. level 2 - case 1
MyTests. level 2 - case 2
MyTests. level 2 - case 1
MyTests. level 2 - case 2
```

I would like to see this: 
```
MyTests. level 1.1. level 2 - case 1
MyTests. level 1.1. level 2 - case 2
MyTests. level 1.2. level 2 - case 1
MyTests. level 1.2. level 2 - case 2
```

**Comment by sksamuel** [comment]
Which report - xml, html, gradle..

**Comment by BorzdeG** [comment]
gradle, html

... I'll add a working example with code to the tests later.

**Comment by sksamuel** [comment]
Do you have time to add the working example ?

**Comment by JesusMcCloud** [comment]
I might be able to provide some more context.

Using Kotest 5.1.0 on a kotlin multiplatform project, including an Android and an iOS target and `FreeSpec` tests, some of which contain `withData` tests.

On the JVM targets (Android and JVM, JUnit 5 runner), the behaviour is exactly as described here (only the `nameFn` result shows up as test name, and the `FreeSpec` container is gone. So what shows up in the test report (and on stdout) is `package.ClassName.nameFnOutput`, which also means that our CI reports are missing lots of rests. The reason for this being that `ClassName` declares lots of `FreeSpec` containers, but each of them uses the same input for a single nested `wichData test series`.

The iOS tests, however, are reported correctly as `package.ClassName.FreeSpecContainerText.nameFnOutput` and show up fine in the test reports.

Now I've read that this is a gradle issue, but how can this be if it works for the iOS target, but not with the Junt5 runner?

**Comment by sksamuel** [comment]
JVM and non-JVM go through different gradle code paths (junit platform vs jetbrains format test output).
I'll try to reproduce this.

**Comment by sksamuel** [comment]
@BorzdeG your issue here I think is covered by these docs.
https://kotest.io/docs/framework/test_output.html

@JesusMcCloud your issue is covered in #2848 I think

**Comment by BorzdeG** [comment]
checked on kotest 5.2.1 - it's fixed!
The output is not the same as in the documentation :)

```
MyTests > level 1.1 > MyTests.level 2 - case 1 PASSED
MyTests > level 1.1 > MyTests.level 2 - case 2 PASSED
MyTests > level 1.2 > MyTests.level 2 - case 1 PASSED
MyTests > level 1.2 > MyTests.level 2 - case 2 PASSED
```

According to the documentation, the output should have looked like this:
```
MyTests. level 1.1 - level 2 - case 1 PASSED
MyTests. level 1.1 - level 2 - case 2 PASSED
MyTests. level 1.2 - level 2 - case 1 PASSED
MyTests. level 1.2 - level 2 - case 2 PASSED
```

**Comment by sksamuel** [comment]
Where does the DemoTest come into it ?

**Comment by BorzdeG** [comment]
> Where does the DemoTest come into it ?

did not correct the output to the log regarding the class name - fixed

**Comment by sksamuel** [comment]
So the class name is included both times? Did you upgrade gradle or a gradle plugin ?

**Comment by BorzdeG** [comment]
kotest: 5.2.1
Gradle: 7.4.1
JVM: 17

> So the class name is included both times?

Yes

PS: Also, the delimiters are different from the documentation

**Comment by sksamuel** [comment]
Not tried gradle 7.4.x yet so maybe they finally fixed some junit platform bugs.

**Comment by hosswald** [comment]
I can confirm from #3068 that it's broken in Gradle 6 but works in Gradle 7 with otherwise similar dependencies (io.kotest:kotest-runner-junit5:5.4.2).

**Comment by Kantis** [comment]
@sksamuel think we can close this, since it seems to affect old combinations of gradle/kotest?

**Comment by sksamuel** [comment]
Yep I think so.

**Comment by Zordid** [comment]
This is sadly broken again with Gradle 8.9... My test names as Gradle sees them do not include the context... :( Are you aware of this?
Gradle 8.9, Kotest 5.9.1

**Comment by Zordid** [comment]
I am using FunSpec with context("...") and test("...")...

The XML test report reads like this:
```
  <testcase name="can be compiled" classname="TraLaTestRunner" time="1.708"/>
  <testcase name="input.in.json transforms to input.out.json" classname="TraLaTestRunner" time="0.0"/>
  <testcase name="can be compiled" classname="TraLaTestRunner" time="0.232"/>
  <testcase name="fails.in.json transforms to fails.out.failure" classname="TraLaTestRunner" time="0.0"/>
  <testcase name="succeeds.in.json transforms to succeeds.out.json" classname="TraLaTestRunner" time="0.0"/>
  <testcase name="can be compiled" classname="TraLaTestRunner" time="0.22"/>
  <testcase name="tacho.in.json transforms to tacho.out.xml" classname="TraLaTestRunner" time="0.0"/>
  <testcase name="can be compiled" classname="TraLaTestRunner" time="0.858"/>
  <testcase name="demo.in.json transforms to demo.out.json" classname="TraLaTestRunner" time="0.001"/>
  <testcase name="can be compiled" classname="TraLaTestRunner" time="1.02"/>
  <testcase name="can be compiled" classname="TraLaTestRunner" time="0.807"/>
  <testcase name="asn.in.json transforms to asn.out.json" classname="TraLaTestRunner" time="0.0"/>
  <testcase name="can be compiled" classname="TraLaTestRunner" time="0.639"/>
```

While in IntelliJ you see my contexts in the structure:
![image](https://github.com/user-attachments/assets/3e956072-4f6d-4f1a-911f-849e8fad6590)

**Comment by sksamuel** [comment]
Was the only change bumping gradle @Zordid

**Comment by Zordid** [comment]
To be honest, I never checked before. Got into this because of a new
project where I noticed that there were only very few tests reported.
The contexts / should structure is never seen in my XML test report names -
I tried some kotest properties I found while searching for this problem -
but nothing helps.

Now I attach the context name programmatically to every test name myself...
Not good...

Sam ***@***.***> schrieb am Mo., 12. Aug. 2024, 03:38:
